daemon/graphdriver/devmapper/deviceset.go

```diff
@@ -2308,6 +2308,33 @@ func (devices *DeviceSet) Shutdown(home string) error {
 	return nil
 }
 
+func noRetryPoolFull(info *devInfo) {
+	dmDevicePath, err := os.Readlink(info.DevName())
+	if err != nil {
+		logrus.Warnf("devmapper: readlink failed for device %v:%v", info.DevName(), err)
+		return
+	}
+
+	dmDeviceName := path.Base(dmDevicePath)
+	filePath := "/sys/fs/xfs/" + dmDeviceName + "/error/metadata/ENOSPC/max_retries"
+	maxRetriesFile, err := os.OpenFile(filePath, os.O_WRONLY, 0)
+	if err != nil {
+		// Older kernels don't have this feature
+		if os.IsNotExist(err) {
+			return
+		}
+		logrus.Warnf("devmapper: Failed to open file %v:%v", filePath, err)
+	}
+
+	// Set max retries to 0
+	_, err = maxRetriesFile.WriteString("0")
+	if err != nil {
+		logrus.Warnf("devmapper: Failed to write to file %v:%v", filePath, err)
```

Comment:
If you return above, then I think you need defer file. Close()  after the open error check. So that here you can just return.